firstproject/src/main/java/com/store/manager/Train.java

```diff
@@ -1,80 +1,36 @@
 package com.store.manager;
 
+import java.util.List;
+import java.util.Optional;
 
 public class Train {
 	 
-	private StringBuilder result ;
+	
+	private List<Wagon> wagons ;
 	
     public Train(String hpp) {
-    	result = new StringBuilder();
-    	getTrainFormat(hpp);
+    	wagons = WagonsFactory.createWagons(hpp);
+    	
     }
 
     public String print() {
-       
+    	StringBuilder result = new StringBuilder(wagons.stream().map(a -> a.getNameWagon()).reduce("",(a,b) -> a + "::" + b ));
```

Comment:
use Collectors.joining() instead of reduce

Reply:
this is the new code :
public String print() {
    	return wagons.stream().map(a -> a.getNameWagon()).collect(Collectors.joining("::"));
    }